azure/cargo-check.yml

```diff
@@ -19,6 +19,13 @@ jobs:
     displayName: Check compilation w/ no features
   - script: cargo check --all --bins --examples --tests --all-features
     displayName: Check compilation w/ all features
+  - bash: "[[ -e Cargo.lock ]] && echo '##vso[task.setvariable variable=is_locked]true' || echo '##vso[task.setvariable variable=is_locked]false'"
+    displayName: Are dependencies locked?
+  # we only check with --tests here since we don't care about the actual
+  # compilation, just the dependencies + dev-dependencies
```

Comment:
`--tests` shouldn't even be needed.  I believe if anything needs the lock file calculated then the whole thing is calculated.